Add fastlane's SPM integration instructions (#1044)
fastlane/docs: docs/getting-started/ios/fastlane-swift.md

🎉

## Get Started (SPM) (Beta)

> Swift Package Manager support is included as a preliminary feature that allows the usage of _fastlane_ with other SPM packages.

Reviewer: ```suggestion
Swift Package Manager support is included as a preliminary feature to integrate _fastlane_ with other SPM packages.
```

Reviewer: @minuscorp this _seems_ better to me, but does it accurately depict what you're suggesting?

Author: _fastlane_ is distributed as a package that allows third-party packages to introduce its usage through SPM. Am I explaining well? 😟

Reviewer: This seems like a complicated concept to get across, and my mind is _relatively_ fresh this morning, but I'm lost. Perdón, pero no lo entiendo, a lo mejor es más facil explicar en español primero, y despues en ingles? Me preocupa que alguien, como yo, no lo va a entender.

Author: El paquete _fastlane_ se distribuye como un paquete SPM, de tal forma que otros paquetes puedan crear ejecutables que lo integren. 
Así queda algo más claro? Es que en una línea es un poco complicado por cómo es la integración de SPM en si 😪

Reviewer: ¡Gracias!
```suggestion
Swift Package Manager (SPM) support adds the ability to distribute _fastlane_ as a Swift Package Manager package, which allows other packages to integrate with the toolset.
```
Does that match up with what you're saying?

> Swift Package Manager support is included as a preliminar support which allows the combination of other SPM packages with _fastlane_.

Reviewer: This sentence needs clarification.

Author: Will do will do, my bad 🙇🏻‍♂️

Reviewer 2: I totally thought `preliminar` was an English word 😂 two individuals that speak latin-like languages writing 🇪🇸  + reviewing 🇧🇷 the same code →  💥 😂😂

### Step 1

Create an executable Swift Package Manager project with `swift package init --type executable`.

### Step 2

Add the _fastlane_ dependency to your `Package.swift`.

```swift

> .package(name: "Fastlane", url: "https://github.com/fastlane/fastlane", .from("2.179.0"))

Reviewer 3: Ahh, I just tried to add this line to my own `Package.swift` file and it said it was invalid. Here's what I think the correct syntax needs to be.

```suggestion
.package(name: "Fastlane", url: "https://github.com/fastlane/fastlane", from: "2.179.0")
```

Hope that helps! :)

Author: Yep, I meant `.upToNextMajor(from: "2.179.0")` but it really does not matter, what a day I had🤯

```

### Step 3

Create your Fastfile.swift file in your package and add the desired lanes, as follows.

``` swift
import Fastlane

// Create a class with: 
class FastFile: LaneFile {
    // Your lanes goes here.
}
```

### Step 4

Add an entry point (`@main`) or a `main.swift` file (mandatory for executable SPM packages) and don't forget to start the _fastlane_ runloop as follows:

```swift
import Fastlane

Main().run(with: Fastfile())
```

### Step 5

> Modify the target of your executable to have executable arguments `lane myLane` or add them in the call after making `swift build`.

Reviewer 3: With me still being rather super new to all of this, is there a tutorial elsewhere that might explain how to do this? 😅

Author: If you generate an Xcode project, you can change the arguments being passed into the executable in the target options or just add them into the generated executable as if it was other CLI.

🎉

## Get Started (SPM) (Beta)

Swift Package Manager support is included as a preliminary feature that allows the usage of _fastlane_ with other SPM packages.

### Step 1

Create an executable Swift Package Manager project with `swift package init --type executable`.

### Step 2

Add the _fastlane_ dependency to your `Package.swift`.

```swift
.package(name: "Fastlane", url: "https://github.com/fastlane/fastlane", from: "2.179.0")
```

A full example of a working package description would be the following.

```swift
// swift-tools-version:5.2

import PackageDescription

> let package = Package(

Author: @TimOliver I have updated the step 2 to give a full working example of a package description, see if this helps a bit ❤️
